java/src/main/java/com/logicalclocks/hsfs/TrainingDatasetBundle.java

```diff
@@ -0,0 +1,49 @@
+package com.logicalclocks.hsfs;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.google.common.collect.Lists;
+import lombok.Getter;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class TrainingDatasetBundle {
+
+  @Getter
+  private Integer version;
+  private Map<String, Dataset<Row>> datasetSplits;
+  private Dataset<Row> dataset;
+
+  public TrainingDatasetBundle(Integer version, Dataset<Row> dataset) {
+    this.version = version;
+    this.dataset = dataset;
+  }
+
+  public TrainingDatasetBundle(Integer version,
+      Map<String, Dataset<Row>> datasetSplits) {
+    this.version = version;
+    this.datasetSplits = datasetSplits;
+  }
+
+  @JsonIgnore
+  public Dataset<Row> getDataset() {
+    return dataset;
+  }
+
+  @JsonIgnore
+  public Dataset<Row> getDataset(String split) {
+    return datasetSplits.get(split);
+  }
+
+  @JsonIgnore
+  public List<String> getSplitNames() {
```

Comment:
Wouldn't it be nice to have a getTrainSplit method as well, since we might know the train_split already from the FeatureView? I'd consider this optional